src/api/HttpApi.ts

```diff
@@ -86,17 +86,43 @@ export default abstract class HttpApi {
     return this.loader.load(this.makePath(path, args));
   }
 
+  protected validateConnectionaArgs({ first, after, last, before }: Args) {
+    if (after && before) {
+      throw new TypeError(
+        '`after` and `before` cursors cannot be specified together',
+      );
+    }
+    if (first != null && last != null) {
+      throw new TypeError('`first` and `last` cannot be specified together.');
+    }
+    if (last != null && !before) {
+      throw new TypeError(
+        'Server does not support backwards pagination without a `before` cursor',
+      );
+    }
+  }
+
   async getPaginatedConnection<T>(
     path: string,
-    { after, first, ...args }: Args,
+    connectionsArgs: Args,
   ): Promise<Maybe<PaginationResult<T>>> {
+    this.validateConnectionaArgs(connectionsArgs);
+    const { after, first, before, last, ...args } = connectionsArgs;
+
+    // assume forward pagination
+    const limit = before ? last : first;
+
+    const query: Args = { ...args, limit, pageSize: limit };
+
+    if (before) query.before = before;
+    if (after) query.cursor = after;
```

Comment:
i thought you added an after, for simmetry?

Reply:
i did but this is backwards compatible with endpoints that haven't upgraded FR yet